app/src/test/java/com/nervousfish/nervousfish/service_locator/ServiceLocatorTest.java

```diff
@@ -0,0 +1,65 @@
+package com.nervousfish.nervousfish.service_locator;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+public final class ServiceLocatorTest {
+    private IServiceLocator serviceLocator;
+
+    @Before
+    public void setup() {
+        serviceLocator = new ServiceLocator("foo");
+    }
+    @Test
+    public void testGetAndroidFilesDir() {
+        assertTrue("foo".equals(serviceLocator.getAndroidFilesDir()));
+    }
+
+    @Test
+    public void testGetDatabase() {
+        assertNotNull(serviceLocator.getDatabase());
```

Comment:
But where do you test if the implementation is right? Not in the module tests I think as there we look at the functionality not the return value of the service locator. Of course if the implementation works it kinda signifies that the service locator did a good job but you are never certain...